File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/PropertyGrid/Functionality/ObjectInspection.vb
Namespace CompanionKit.Controls.PropertyGrid.Functionality
	Public Class ObjectInspection

		Public Sub New()

			' This call is required by the Visual WebGui UserControl Designer.
			InitializeComponent()



		End Sub

		''' <summary>
		''' Select the initial and another label to inspection
		''' </summary>
		''' <param name="sender"></param>
		''' <param name="e"></param>
		''' <remarks></remarks>
		Private Sub objMultiple_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles objMultiple.Click
			Me.objPropGrid.SelectedObjects = New Object() {Me.objInitial, Me.objAnotherLabel}
		End Sub

		''' <summary>
		''' Select the link label and change the URL property
		''' </summary>
		''' <param name="sender"></param>
		''' <param name="e"></param>
		''' <remarks></remarks>
		Private Sub objButton03_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles objButton03.Click
			Me.objPropGrid.SelectedObject = Me.objVWGLinkLabel
		End Sub

		''' <summary>
		''' Select the other label to change
		''' </summary>
		''' <param name="sender"></param>
		''' <param name="e"></param>
		''' <remarks></remarks>
		Private Sub objButton02_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles objButton02.Click
			Me.objPropGrid.SelectedObject = Me.objAnotherLabel
		End Sub

		''' <summary>
		''' Select the label to inspection
		''' Change the Text property, Border style, Border Width
		''' </summary>
		''' <param name="sender"></param>
		''' <param name="e"></param>
		''' <remarks></remarks>
		Private Sub objButton01_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles objButton01.Click
			Me.objPropGrid.SelectedObject = Me.objInitial
		End Sub
	End Class

End Namespace

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/PropertyGrid/Functionality/ObjectInspection.cs
#region Using

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using Gizmox.WebGUI.Common;
using Gizmox.WebGUI.Forms;

#endregion

namespace CompanionKit.Controls.PropertyGrid.Functionality
{
    public partial class ObjectInspection : UserControl
    {
        public ObjectInspection()
        {
            InitializeComponent();
        }
        
        /// <summary>
        /// Select the label to inspection
        /// Change the Text property, Border style, Border Width
        /// </summary>
        private void objButton01_Click(object sender, EventArgs e)
        {
            objPropGrid.SelectedObject = objInitial;
        }

        /// <summary>
        /// Select the other label to change
        /// </summary>
        private void objButton02_Click(object sender, EventArgs e)
        {
            objPropGrid.SelectedObject = objAnotherLabel;
        }

        /// <summary>
        /// Select the link label and change the URL property
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        private void objButton03_Click(object sender, EventArgs e)
        {
            objPropGrid.SelectedObject = objVWGLinkLabel; 
        }

        /// <summary>
        /// Select the initial and another label to inspection
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        private void objMultiple_Click(object sender, EventArgs e)
        {
            objPropGrid.SelectedObjects = 
                new object[]
                {
                    objInitial,
                    objAnotherLabel
                };
        }
    }
}